frontend/public/components/dashboards-page/dashboards.tsx

```diff
@@ -2,24 +2,51 @@ import * as React from 'react';
 import { RouteComponentProps } from 'react-router-dom';
 import { connect } from 'react-redux';
 
+import * as plugins from '../../plugins';
 import { OverviewDashboard } from './overview-dashboard/overview-dashboard';
-import { HorizontalNav, PageHeading, LoadingBox } from '../utils';
+import { HorizontalNav, PageHeading, LoadingBox, Page, AsyncComponent } from '../utils';
+import { Dashboard } from '../dashboard/dashboard';
+import { DashboardGrid, GridPosition } from '../dashboard/grid';
+import { DashboardsCard } from '@console/plugin-sdk';
 
-const tabs = [
+const getCardsOnPosition = (cards: DashboardsCard[], position: GridPosition): React.ComponentType<any>[] =>
+  cards.filter(c => c.properties.position === position).map(c => () => <AsyncComponent loader={c.properties.loader} />);
+
+const getPluginTabs = (): Page[] => {
+  const cards = plugins.registry.getDashboardsCards();
+  return plugins.registry.getDashboardsTabs().map(tab => {
+    const tabCards = cards.filter(c => c.properties.tab === tab.properties.id);
+    return {
+      href: tab.properties.id,
```

Comment:
We should also write an extension check that ensures tab IDs are unique, I'll put it on my task list.